src/Workspaces/Remote/ServiceHub/Services/Host/RemoteHostService.cs

```diff
@@ -51,6 +52,14 @@ internal partial class RemoteHostService : ServiceBase, IRemoteHostService, IAss
 
         static RemoteHostService()
         {
+            if (GCSettings.IsServerGC)
+            {
+                // Server GC runs processor-affinitized threads with high priority. To avoid interfering with other
+                // applications while still allowing efficient out-of-process execution, slightly reduce the process
+                // priority when using server GC.
+                Process.GetCurrentProcess().PriorityClass = ProcessPriorityClass.BelowNormal;
```

Comment:
This will most likely throw on Mono. Since we're integrating remote workspace services in VSMac, could you please apply the changes that were done in https://github.com/dotnet/roslyn/pull/47373 ?

Reply:
➡️ Restored ProcessExtensions.

Reply:
Thanks!

Reply:
Do we use the Banned API analyzer? Perhaps we should ban `PriorityClass` to catch this in future